mm/kfence.c

```diff
@@ -254,18 +262,20 @@ void *guarded_alloc(size_t size, gfp_t gfp)
 		item = list_entry(kfence_freelist.list.next,
 				  struct kfence_freelist_t, list);
 		obj = item->obj;
-		kfence_freelist.list.next = item->list.next;
+		list_del(&(item->list));
```

Comment:
list_del followed by list_add is undefined. If you turn on CONFIG_DEBUG_LIST you will likely get warnings.

I think this needs to be list_del_init

Reply:
I don't think this is necessary as long as the lists are locked. See e.g. https://marc.info/?l=linux1394-devel&m=114328842931957
CONFIG_DEBUG_LIST also didn't show any warnings.